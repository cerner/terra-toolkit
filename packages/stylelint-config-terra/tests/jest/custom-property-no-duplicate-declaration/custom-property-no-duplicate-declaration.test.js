/**
 * @jest-environment node
 */
const stylelint = require('stylelint');
const path = require('path');

const config = {
  plugins: [
    path.resolve(__dirname, '..', '..', '..', 'src', 'rules', 'custom-property-no-duplicate-declaration', 'custom-property-no-duplicate-declaration'),
  ],
  rules: {
    'terra/custom-property-no-duplicate-declaration': true,
  },
};

describe('custom-property-no-duplicate-declaration', () => {
  it('does not error with a one custom property', () => {
    const result = stylelint.lint({
      code: 'a { color: var(--test-color, #000); }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with two distinct custom properties', () => {
    const result = stylelint.lint({
      code: 'a { color: var(--test-color, #000); } .anchor { color: var(--test-color, #000); }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does error with sibling duplicate custom properties', () => {
    const result = stylelint.lint({
      code: 'a { color: var(--test-color, #000); } .anchor { color: var(--test-color, #111); }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does error with reversed sibling duplicate custom properties', () => {
    const result = stylelint.lint({
      code: '.anchor { color: var(--test-color, #111); } a { color: var(--test-color, #000); }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does error with nested duplicate custom properties', () => {
    const result = stylelint.lint({
      code: '.anchor { color: var(--test-color, #111); a { color: var(--test-color, #000); } }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });
});
