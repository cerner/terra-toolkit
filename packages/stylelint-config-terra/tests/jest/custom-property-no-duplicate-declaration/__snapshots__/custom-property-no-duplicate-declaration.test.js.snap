// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-property-no-duplicate-declaration does error with nested duplicate custom properties 1`] = `
Array [
  Object {
    "column": 11,
    "line": 1,
    "rule": "terra/custom-property-no-duplicate-declaration",
    "severity": "error",
    "text": "Unexpected duplicate property declaration. --test-color has already been declared as #000. (terra/custom-property-no-duplicate-declaration)",
  },
]
`;

exports[`custom-property-no-duplicate-declaration does error with reversed sibling duplicate custom properties 1`] = `
Array [
  Object {
    "column": 11,
    "line": 1,
    "rule": "terra/custom-property-no-duplicate-declaration",
    "severity": "error",
    "text": "Unexpected duplicate property declaration. --test-color has already been declared as #000. (terra/custom-property-no-duplicate-declaration)",
  },
]
`;

exports[`custom-property-no-duplicate-declaration does error with sibling duplicate custom properties 1`] = `
Array [
  Object {
    "column": 49,
    "line": 1,
    "rule": "terra/custom-property-no-duplicate-declaration",
    "severity": "error",
    "text": "Unexpected duplicate property declaration. --test-color has already been declared as #000. (terra/custom-property-no-duplicate-declaration)",
  },
]
`;
